### DISCUSSION TITLE
Need context around how the project is compiling the dependencies eg. in `@umijs/utils`

### DISCUSSION THREAD

**Discussion by kuldeepat1lint** [umijs/umi#11428] [Q&A]
So I have explored some of the project noticed u guys are using bundleDeps script to compile the dependencies with `pnpm run build:deps` which compiles the dependencies mentioned in the package.json file's `compiledConfig.deps` field.

I have tried to run the same script on fresh clone of the project but it's failing and seemingly different output from current ones. For instance I have just removed the compiled folder from `@umijs/utils` and ran the script but it's failing with 'prettier has "node:"' so i change the script for node 14 patch for this only, this time it ran but its output in compiled folder is different from current one.

Am I missing something here? I am trying to understand how the project is compiling the dependencies and how can I do the same for my project.

**Comment by fz6m** [comment]
Check your env version matches:

https://github.com/umijs/umi/blob/c31259b13f71f7c649b2b9796d7253ef360e4d44/package.json#L123-L127

**e.g.**  recompile the `compiled/axios` deps of the `@umijs/utils`:

```bash
  pnpm i
  cd ./packages/utils
  rm -rf ./compiled/axios
  pnpm build:deps --dep axios
```

**Reply by kuldeepat1lint** [reply]
@fz6m 

node: v18.17.0
pnpm: 8.6.2

diff:: https://github.com/kuldeepat1lint/umi/commit/e0e8c20f49d79bb57c9e516518b67f508d050b40

I removed zod and axios from freshly clone repo and ran `pnpm build:deps --dep axios` && `pnpm build:deps --dep zod` the axios compiled differently than in git, where zod complies the same, but its types were not included, which are in zod's lib folder. I still don't get it, how the project compiles its deps...?
